docs/customizing.rst

```diff
@@ -0,0 +1,198 @@
+..
+    This file is part of Invenio.
+    Copyright (C) 2020 CERN.
+
+    Invenio is free software; you can redistribute it and/or modify it
+    under the terms of the MIT License; see LICENSE file for more details.
+
+Customizing
+===========
+
+
+how to inject/override the Invenio config (url, facets, the DOM id where to render the app): via parametrize or remove the Jinja macro and inject JSON config
+
+Configuring the settings of your Search Application
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A helper is provided to generate the configuration needed for the
+React-Searchkit application.
+This configuration contains the following information:
+
+`aggs`: a list of aggregations that will add accordingly facets to your search.
+
+.. code-block:: python
+
+    [
+        {
+            "aggName": "access_right",
+            "field": "access_right",
+            "title": "Access Right"
+        }
+    ]
+
+`appId`: the value here is needed since it will be used internally to
+differentiate the resources of multiple apps.
+"rdm-search"
+`initialQueryState`: This object is used to compose the initial search request
+that will be sent to the backend.
+
+.. code-block:: python
+
+    {
+        "hiddenParams": null,
+        "size": 10
+    }
+
+`layoutOptions`: here we can choose which layouts are going to be enabled
+in our search page, if both are present then a layout switcher will also be
+present in the UI.
+
+.. code-block:: python
+
+    {
+    "gridView": false,
+    "listView": true
+    }
+
+`paginationOptions`: By changing this object you can increase the available
+results per page.
+
+.. code-block:: python
+
+    {
+    "defaultValue": 10,
+    "resultsPerPage": [
+      {
+        "text": "10",
+        "value": 10
+      },
+      {
+        "text": "20",
+        "value": 20
+      }]
+      }
+
+`searchApi`: the searchApi defines the URL for the list REST API, as well as
+any headers that may need to be attached to the request.
+
+.. code-block:: python
+
+    {
+    "axios": {
+      "headers": {
+        "Accept": "application/vnd.inveniordm.v1+json"
+      },
+      "url": "/api/records",
+      "withCredentials": true
+    }
+    }
+
+`sortOptions`
+
+.. code-block:: python
+
+    [
+        {
+        "sortBy": "bestmatch",
+        "text": "Best match"
+        },
+        {
+        "sortBy": "newest",
+        "text": "Newest"
+        }
+    ]
+
+This helper needs to be provided with the ID of the endpoint registered in your
+Invenio-Records-REST configuration.
+https://github.com/inveniosoftware/invenio-records-rest/blob/027794e6ae8dbe2448c59a64b30cdd3c4abb7eb7/invenio_records_rest/config.py#L26
```

Comment:
```suggestion
`Invenio-Records-REST configuration. <https://github.com/inveniosoftware/invenio-records-rest/blob/027794e6ae8dbe2448c59a64b30cdd3c4abb7eb7/invenio_records_rest/config.py#L26>`_
```

Reply:
So this was something that I wanted feedback on, is it valid to point to the code of another repo?
Since this will be a permalink to the current version.

Reply:
I don't really like it but:
1- we need it
2- Better to be hyperlink than a full link (long an ugly) in the docs...